pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,128 @@ def wrapper(*args, **kwargs):
         )
 
     return wrapper
+
+
+@functools.lru_cache
+def _get_cond_qfunc_prim():
+    """Get the cond primitive for quantum functions."""
+
+    # JAX should be installed if capture is enabled
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = jax.core.Primitive("cond")
+    cond_prim.multiple_results = True
+
+    @cond_prim.def_impl
+    def _(condition, elifs_conditions, *args, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        def run_jaxpr(jaxpr, *args):
+            return jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
+
+        def true_branch(args):
+            return run_jaxpr(jaxpr_true, *args)
+
+        def elif_branch(args, elifs_conditions, jaxpr_elifs):
+            if not jaxpr_elifs:
+                return false_branch(args)

```

Comment:
> since I wouldn't expect the recursion depth to be very large.

Exactly, I don't expect that the stack allocation ever becomes a problem for the number of `elifs` conditions that we usually provide to this function